contracts/FIFOMarket.sol

```diff
@@ -136,9 +160,17 @@ contract FIFOMarket is
   ) public override returns (bytes4) {
     for (uint256 i = 0; i < ids.length; i++) {
       uint256 removalAmount = _removal.balanceOf(address(this), ids[i]);
-      totalSupply += removalAmount;
-      _queue[_queueNextInsertIndex] = (ids[i]);
-      _queueNextInsertIndex++;
+      totalActiveSupply += removalAmount;
+      totalNumberActiveRemovals += 1;
+      address supplierAddress = ids[i].supplierAddress();
```

Comment:
since `_addActiveSupplier` mutates similar state but is not called if they are already an active supplier, I'm thinking this should probably be extracted to something like `_updateActiveSupply`